smt/surrogate_models/tests/test_krg_based.py

```diff
@@ -176,7 +172,7 @@ def test_kpls_mixed_4D(self):
         for i in range(x_out.shape[0]):
             if abs(float(x_out[i, :][2]) - int(float(x_out[i, :][2]))) > 10e-8:
                 Equality_check = False
-            if not (x_out[i, :][1] == "blue" or x_out[i, :][1] == "red"):
+            if not (x_out[i, :][1] == 0 or x_out[i, :][1] == 1):
```

Comment:
No string outputs but string inputs.  Categorical inputs are not taken into consideration anymore.